libminifi/src/core/state/nodes/ResponseNodeLoader.cpp

```diff
@@ -238,6 +242,10 @@ std::vector<SharedResponseNode> ResponseNodeLoader::loadResponseNodes(const std:
   }
 
   for (const auto& response_node : response_nodes) {
+    std::lock_guard<std::mutex> guard(initialization_mutex_);
```

Comment:
Please move the lock guard outside of the loop. Letting go and reacquiring the same mutex leads to scheduling inefficiencies if there is any contention.